indexing-hadoop/src/main/java/org/apache/druid/indexer/Utils.java

```diff
@@ -143,4 +151,66 @@ public static String getFailureMessage(Job failedJob, ObjectMapper jsonMapper)
       return null;
     }
   }
+
+  /**
+   * When the JobHistory server is unavailable, it's possible to fetch the application's status
+   * from the YARN ResourceManager instead.
+   */
+  public static boolean checkAppSuccessFromYarnRM(Job job) throws Exception
+  {
+    log.info("Could not retrieve job status from JobHistory server, trying YARN ResourceManager instead.");
+    final HttpClient httpClient = new HttpClient();
+    final AtomicBoolean succeeded = new AtomicBoolean(false);
+    try {
+      httpClient.start();
+      RetryUtils.retry(
+          () -> {
+            checkAppSuccessFromYarnRMOnce(httpClient, job, succeeded);
+            return null;
+          },
+          ex -> {
+            return !succeeded.get();
+          },
+          5
+      );
+      return succeeded.get();
+    }
+    catch (Exception e) {
+      log.error(e, "Got exception while trying to contact YARN RM.");
+      // we're already in a best-effort fallback failure handling case, just stop if we have issues with the http client
+      return false;
+    }
+    finally {
+      httpClient.stop();
+    }
+  }
+
+  private static void checkAppSuccessFromYarnRMOnce(
+      HttpClient httpClient,
+      Job job,
+      AtomicBoolean succeeded
+  ) throws IOException, InterruptedException, ExecutionException, TimeoutException
+  {
+    log.info("Attempting to retrieve app status from YARN RM.");
+    String appId = StringUtils.replace(job.getJobID().toString(), "job", "application");
+    String yarnRM = job.getConfiguration().get("yarn.resourcemanager.webapp.address");
+    log.info("YARN RM address: " + yarnRM);
```

Comment:
maybe combine log at 194 and 197 into single message and use formatter?

```suggestion
    log.info("Attempting to retrieve app status from YARN RM [%s].", yarnRM);
```

Reply:
Removed these and added a new log that prints out the full endpoint URL

Reply:
even better